pkg/planner/cascades/old/transformation_rules.go

```diff
@@ -1327,6 +1316,7 @@ func (*PushTopNDownProjection) OnTransform(old *memo.ExprIter) (newExprs []*memo
 	for i := len(newTopN.ByItems) - 1; i >= 0; i-- {
 		switch newTopN.ByItems[i].Expr.(type) {
 		case *expression.Constant, *expression.CorrelatedColumn:
+			// TODO: investigate why we cannot use: topN.ByItems = slices.Delete(newTopN.ByItems, i, i+1)
```

Comment:
Very interesting issue. May I ask what happens if we change it here?

I suspect it might be caused by the behavior shown in this example: https://play.golang.com/p/Tem8vEmY5U3.

Reply:
Oh, I missed that it was not the same slice that was assigned and appended... Good catch. I will see if I can fix it.

Reply:
We should probably use slices.DeleteFunc() instead of looping and deleting in the loop.

Reply:
@Rustin170506 Should `topN.ByItems` also be set? Or is that an unknown bug since before?